src/upload.d

```diff
@@ -63,25 +63,36 @@ struct UploadSession
 				log.vlog("The file does not exist anymore");
 				return false;
 			}
-			// request the session status
-			JSONValue response;
+			// Can we read the file - as a permissions issue or file corruption will cause a failure on resume
+			// https://github.com/abraunegg/onedrive/issues/113
 			try {
-				response = onedrive.requestUploadStatus(session["uploadUrl"].str);
-			} catch (OneDriveException e) {
-				if (e.httpStatusCode == 400) {
-					log.vlog("Upload session not found");
+				// attempt to read the first 10MB of the file
```

Comment:
perhaps this test should be consolidated somewhere in a utility class to avoid duplication?